app/controllers/public_controller.rb

```diff
@@ -21,16 +21,49 @@ def privacy_policy
   end
 
   def check_if_saw_cookie_notice
-    flash[:info] = "This site uses cookies. See our <a href='/cookie-policy'>Cookie Policy</a> for details.".html_safe unless cookies[:saw_cookie_notice]
+    unless cookies[:saw_cookie_notice]
+      flash[:info] = "This site uses cookies. See our <a href='/cookie-policy'>Cookie Policy</a> for details.".html_safe
+    end
   end
 
   def cookie_policy
     @title = t('public.cookie_policy.title')
   end
 
+  def contact_us
+    if request.post?
+      @contact_request = ContactRequest.new(contact_us_params)
+      if verify_hcaptcha(model: @contact_request) && @contact_request.save
+        flash[:notice] = 'Your request has been saved. We will contact you ASAP.'
+        send_email_to_admins_about_new_request
+      else
+        render 'contact_us'
+      end
+    else
+      @contact_request = ContactRequest.new
+    end
+    @title = t('public.contact_us.title')
+  end
+
   def saw_cookie_notice
     cookies[:saw_cookie_notice] = true
-    flash[:info] = nil
+    flash.clear
```

Comment:
We should clear just the cookie notice, there might be some other flash messages that'll get fully wiped out this way